### PR TITLE
Adding cache clear mechanism

### PR DESCRIPTION
**Pull request by jiwalker-usgs**

### CODE REVIEW

File: coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ */
+@Path(DataURI.CACHE_PATH)
+@PermitAll
+public class CacheResource {
+	
+	private static final Logger log = LoggerFactory.getLogger(CacheResource.class);
+
+	private static final String CACHE_NAME_PROP = "coastal-hazards.portal.geoserver.cache.name";
+	private static final String CACHE_NAME = "proxyCache";
+	private static final String CACHE_LOCATION_PROP = "coastal-hazards.portal.geoserver.cache.name";
```
Comment (isuftin) [inline coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java]: Why two constants with the same value? Are you looking to make them different at some point in the future?

File: coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	private static final String CACHE_LOCATION = "${java.io.tmpdir}/ehcache";
```
Comment (isuftin) [inline coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java]: How does ${} work in here?

File: coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+					.dynamicConfig(false)
+					.cache(ehConfig);
+			CacheManager cacheManager = CacheManager.create(managerConfig);
+			Ehcache ehcache = cacheManager.getEhcache(cacheName);
+			if (ehcache != null) {
+				ehcache.flush();
+				ehcache.removeAll();
+				response = Response.ok().build();
+			} else {
+				response = Response.serverError().entity("Error location cache to clear").build();
```
Comment (isuftin) [inline coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java]: What does this error mean?

**Comment by isuftin** [comment]
This looks good to ship, but are you able to unit test this functionality? It seems like CacheResource.java L51-70 is very testable.

File: coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java (hunk 1 above)
Comment (jiwalker-usgs) [inline coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java]: oops, this should be cache.location

File: coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java (hunk 2 above)
Comment (jiwalker-usgs) [inline coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java]: ehcache expands it

File: coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java (hunk 3 above)
Comment (jiwalker-usgs) [inline coastal-hazards-portal/src/main/java/gov/usgs/cida/coastalhazards/rest/data/CacheResource.java]: typo, meant locating

**Comment by jiwalker-usgs** [comment]
I'll try to make a unit test.  This is jumping on a cache that is built elsewhere, which is the real trick here, but I should be able to make a cache and clear it.

**Comment by jiwalker-usgs** [comment]
Test is fragile because the cache location needs to be the default in the cache clear case.  We should probably move to a central place where we set the cache location and use that for all usages of the cache.

**Pull request merged**